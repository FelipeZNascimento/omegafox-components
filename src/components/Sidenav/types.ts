import { TNavbarThemes } from 'components/Navbar/types';

export type TSidenavButton = {
  id: number;
  text: string;
  url: string;
  withCredentials?: boolean;
  renderingFunction?: () => JSX.Element;
};

export interface ISidenavProps {
  isOpen: boolean;
  logo?: string;
  renderBottom?: () => JSX.Element;
  selectedId: number;
  sidenavButtons: TSidenavButton[];
  theme?: TNavbarThemes;
  onClick: (navbarButton: TSidenavButton) => void;
  onClose: () => void;
}
