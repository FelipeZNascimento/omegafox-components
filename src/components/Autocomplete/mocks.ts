import { TDropdownItem } from './types';

export const dropdownList: TDropdownItem[] = [
  {
    id: 0,
    name: 'Lionel Messi',
    abbreviationEn: 'ARG'
  },
  {
    id: 1,
    name: 'Cristiano Ronaldo',
    abbreviationEn: 'POR'
  },
  {
    id: 2,
    name: 'Mbappé',
    abbreviationEn: 'FRA'
  },
  {
    id: 3,
    name: 'Caio Jr.',
    abbreviationEn: 'BRA'
  },
  {
    id: 4,
    name: 'Darwin Nuñez',
    abbreviationEn: 'URU'
  }
];
