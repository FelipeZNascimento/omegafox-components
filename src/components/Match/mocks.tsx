import { IMatchProps, ITeamProps, TClockFootball } from '../types';
import { FOOTBALL_MATCH_STATUS } from '../Clock/constants';

const matchId = 0;
export const teamLeft: ITeamProps = {
  id: 0,
  align: 'left',
  colors: ['#FFDC02', '#19AE47'],
  isEditable: false,
  isForceMobile: false,
  logo: 'https://teamcolorcodes.com/wp-content/uploads/2021/12/Brazil-National-Football-Team-Logo-211x300.png',
  matchId: matchId,
  name: 'Brasil',
  nameShort: 'BRA',
  score: 4
};

export const teamRight: ITeamProps = {
  id: 1,
  align: 'right',
  colors: ['#FFFFFF', '#43A1D5'],
  isEditable: false,
  isForceMobile: false,
  logo: 'https://teamcolorcodes.com/wp-content/uploads/2021/12/Argentina-National-Football-Team-Logo-192x300.png',
  matchId: matchId,
  name: 'Argentina',
  nameShort: 'ARG',
  score: 0
};

export const teamLeftNoBet: ITeamProps = {
  ...teamLeft,
  score: null
};

export const teamRightNoBet: ITeamProps = {
  ...teamRight,
  score: null
};

export const footballClock: TClockFootball = {
  time: 45,
  status: FOOTBALL_MATCH_STATUS.FIRST_HALF
};

export const matchInfo: IMatchProps = {
  id: matchId,
  isExpandable: false,
  clock: footballClock,
  expandableContent: () => {
    return <p>Vila Capanema</p>;
  },
  sport: 'football',
  teams: [teamLeft, teamRight],
  timestamp: 1662682800
};

export const matchInfoNoBets: IMatchProps = {
  ...matchInfo,
  teams: [teamLeftNoBet, teamRightNoBet]
};
