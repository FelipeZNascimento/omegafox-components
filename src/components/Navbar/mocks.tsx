import { TNavbarButton } from './types';

export const navbarButtons: TNavbarButton[] = [
  {
    id: 0,
    text: 'InÃ­cio',
    url: ''
  },
  {
    id: 1,
    text: 'Resultados',
    url: ''
  },
  {
    id: 2,
    text: 'Apostar',
    url: ''
  },
  {
    id: 3,
    text: 'Regras',
    url: ''
  },
  {
    id: 4,
    text: 'PrÃªmios',
    url: ''
  },
  {
    id: 5,
    text: 'Extras',
    url: ''
  },
  {
    id: 6,
    text: '',
    url: '',
    renderingFunction: () => {
      return <a href="#">link</a>;
    }
  }
];
