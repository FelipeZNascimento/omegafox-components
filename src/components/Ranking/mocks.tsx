import { IRankingProps, TRankingRow } from './types';
const singleRow: TRankingRow = [
  { id: 0, renderingFunction: () => <div>01.</div> },
  {
    id: 1,
    renderingFunction: () => <div>Felipe Zanon Felipe Zanon Felipe Zanon</div>
  },
  { id: 2, renderingFunction: () => <div>999</div> }
];

export const tableConfig: IRankingProps = {
  isHeader: true,
  columns: [
    {
      id: 0,
      align: 'left',
      flex: 1,
      renderingFunction: () => <div>Pos</div>
    },
    {
      id: 1,
      align: 'left',
      flex: 4,
      renderingFunction: () => <div>Nome</div>
    },
    {
      id: 2,
      align: 'right',
      flex: 1,
      renderingFunction: () => <div>Pts.</div>
    }
  ],
  rows: [singleRow, singleRow, singleRow, singleRow, singleRow, singleRow]
};
