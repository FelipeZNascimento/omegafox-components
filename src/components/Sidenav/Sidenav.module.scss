@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.darkBackground {
  height: 100vh;
  position: absolute;
  top: 0;
  width: 100vw;
  z-index: $always-on-top + 1;

  &Open {
    @include bg-rgba-colored($color-black, 0.2);

    left: 0;
    transition: background-color $transition-time, left $transition-time;
  }

  &Closed {
    @include bg-rgba-colored($color-black, 0);

    left: -100vw;
    transition: background-color $transition-time, left $transition-time;
  }
}
.container {
  @include shadowed(0.2);
  @include bg-colored($color-grey1);
  @include font-sized(l2);
  @include flex-aligned(center, center);

  flex: 1 0 auto;
  flex-direction: column;
  height: 100%;
  position: fixed;
  top: 0;
  z-index: $always-on-top + 1;

  &Open {
    margin-left: 0;
    transition: margin $transition-time;
  }

  &Closed {
    margin-left: -50%;
    transition: margin $transition-time;
  }
}

.logoContainer {
  @include flex-aligned(center, flex-start);
  @include padded(vertical, s);

  flex: 0;
  max-width: 100px;
  max-height: 100px;

  .logo {
    height: 100%;
    width: auto;
  }
}

.nav {
  @include padded(vertical, l);
  @include padded(horizontal, l);
  @include flex-aligned(center, flex-start);
  flex: 1;
  flex-direction: column;

  button,
  input[type="submit"],
  input[type="reset"] {
    border: none;
    cursor: pointer;
    outline: inherit;
  }

  .navButton {
    @include bg-colored($color-grey1);
    @include font-colored($color-grey4);
    @include padded(all, m);

    border-radius: 16px;
    font: inherit;
    transition: all $transition-time;
    width: 100%;

    &Selected {
      @include font-colored($color-qatar-red);
    }

    &:hover {
      @include bg-colored($color-grey2);

      transition: all $transition-time;
    }
  }
}

.bottom {
  @include flex-aligned(center, flex-start);
  @include padded(vertical, s);

  flex: 0;
}
