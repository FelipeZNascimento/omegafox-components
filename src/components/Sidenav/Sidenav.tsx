import { useRef, useEffect } from 'react';
import classNames from 'classnames';

import { Backdrop } from '../index';
import { ISidenavProps } from './types';
import styles from './Sidenav.module.scss';

export const Sidenav = ({
  isOpen,
  logo,
  renderBottom,
  selectedId,
  sidenavButtons,
  theme = 'copa',
  onClick,
  onClose
}: ISidenavProps) => {
  const titleCase = (string: string) => {
    return string[0].toUpperCase() + string.slice(1).toLowerCase();
  };

  const classTheme = `navButton${titleCase(theme)}`;
  const containerClass = classNames(styles.container, {
    [styles.containerOpen]: isOpen,
    [styles.containerClosed]: !isOpen
  });

  //Hook that alerts clicks outside of the passed ref
  function useOutsideAlerter(ref: React.RefObject<HTMLDivElement>) {
    useEffect(() => {
      // Alert if clicked on outside of element
      function handleClickOutside(event: MouseEvent) {
        if (ref.current && !ref.current.contains(event.target as Node)) {
          onClose();
        }
      }
      // Bind the event listener
      document.addEventListener('mousedown', handleClickOutside);
      return () => {
        // Unbind the event listener on clean up
        document.removeEventListener('mousedown', handleClickOutside);
      };
    }, [ref]);
  }

  const renderButtons = () => {
    return sidenavButtons.map((item) => {
      if (item.renderingFunction) {
        return (
          <button
            key={item.id}
            className={styles[classTheme]}
            onClick={() => onClick(item)}
          >
            {item.renderingFunction()}
          </button>
        );
      }

      const buttonClass = classNames(styles[classTheme], {
        [styles[`${classTheme}Selected`]]: selectedId === item.id
      });

      const onButtonClick = () => {
        onClick(item);
      };

      return (
        <button key={item.id} className={buttonClass} onClick={onButtonClick}>
          {item.text}
        </button>
      );
    });
  };

  const wrapperRef = useRef(null);
  useOutsideAlerter(wrapperRef);

  return (
    <Backdrop align="left" isOpen={isOpen}>
      <div
        className={containerClass}
        ref={wrapperRef}
        data-testid="test-sidenav__container"
      >
        {logo && (
          <div className={styles.logoContainer}>
            <img alt="Logo" className={styles.logo} src={logo} />
          </div>
        )}
        <nav className={styles.nav}>{renderButtons()}</nav>
        {renderBottom && <div className={styles.bottom}>{renderBottom()}</div>}
      </div>
    </Backdrop>
  );
};
