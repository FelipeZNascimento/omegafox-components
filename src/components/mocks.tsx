export {
  navbarButtonsLeft,
  navbarButtonsRight,
  navbarRenderButton
} from './Navbar/mocks';
export {
  footballClock,
  matchInfoNoBets,
  matchInfo,
  teamLeft,
  teamRight
} from './Match/mocks';
export { tableConfig } from './Ranking/mocks';
export { selectorItems } from './Selector/mocks';
