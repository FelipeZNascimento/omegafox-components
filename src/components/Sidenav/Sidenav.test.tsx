import React from 'react';
import { render, waitFor } from '@testing-library/react';

import { Sidenav } from './Sidenav';
// import { ISidenavProps } from '../types';
import { navbarButtonsLeft as sidenavButtons } from '../mocks';

describe('SidebarComponent', () => {
  // const renderComponent = ({ isOpen, sidenavButtons }: ISidenavProps) =>
  //   render(
  //     <Sidenav
  //       isOpen={isOpen}
  //       selectedId={sidenavButtons[0].id}
  //       sidenavButtons={sidenavButtons}
  //       onClick={jest.fn}
  //       onClose={jest.fn}
  //     />
  //   );

  it('should render', async () => {
    const { container, rerender, getByTestId } = render(
      <Sidenav
        isOpen={false}
        selectedId={sidenavButtons[0].id}
        sidenavButtons={sidenavButtons}
        onClick={jest.fn}
        onClose={jest.fn}
      />
    );

    rerender(
      <Sidenav
        isOpen={true}
        selectedId={sidenavButtons[0].id}
        sidenavButtons={sidenavButtons}
        onClick={jest.fn}
        onClose={jest.fn}
      />
    );

    await waitFor(() => {
      const sidenavContainer = getByTestId('test-sidenav__container');
      expect(sidenavContainer).toBeVisible();
    });

    expect(container).toMatchSnapshot();
  });
});
