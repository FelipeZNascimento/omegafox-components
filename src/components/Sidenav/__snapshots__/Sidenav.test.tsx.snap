// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarComponent should render 1`] = `
<div>
  <div
    class="darkBackground darkBackgroundOpen"
  >
    <div
      class="container containerOpen"
    >
      <div
        class="logoContainer"
      >
        <img
          class="logo"
          src="https://upload.wikimedia.org/wikipedia/en/thumb/e/e3/2022_FIFA_World_Cup.svg/1200px-2022_FIFA_World_Cup.svg.png"
        />
      </div>
      <nav
        class="nav"
      >
        <button
          class="navButton navButtonSelected"
        >
          Início
        </button>
        <button
          class="navButton"
        >
          Resultados
        </button>
        <button
          class="navButton"
        >
          Apostar
        </button>
        <button
          class="navButton"
        >
          Regras
        </button>
        <button
          class="navButton"
        >
          Prêmios
        </button>
        <button
          class="navButton"
        >
          Extras
        </button>
        <button
          class="navButton"
        >
          <a
            href="#"
          >
            link
          </a>
        </button>
      </nav>
    </div>
  </div>
</div>
`;
